### PR TITLE
Adding SMTP username and password parameters for the authentication

### PR DESCRIPTION
**Pull request by makirill**
Some SMTP servers (e.g. smtp.gmail.com) requires account authentication with password. Two optional environment variable `SMTP_USER` and `SMTP_USER_PASSWORD` were added in order to support it.

The `SMTP_TLS_CERT` and `SMTP_TLS_KEY` environment variables are not required for `smtp.gmail.com` server and they were made optional.

Helm Chart was updated to configure SMTP server accounts details e.g.:
```
smtp:
  account:
    generateSecret: true
    secretName: smtp-secret
    username: login@smtp-mail.org
    password: secret_password
```
where:
- `account` - Specifies account details for SMTP server authentication (optional)
- `account.generateSecret` - Whether new secret with SMTP username/password will be generated. Will try to mount secret with "secretName" if generateSecret == false
- `account.secretName` - Override default secret name (chart name + '-smtp-credentials')
- `account.username` - Account used for the SMTP server authentication
- `account.password` - Password used for the SMTP Server authentication

### CODE REVIEW

File: notifier/operator/operator.py [hunk 1]
```diff
@@ -54,19 +56,17 @@
...
 
 # Write cert and key to tmp file and load
 if smtp_tls_cert:
     smtp_tls_cert_fd, smtp_tls_cert_file = mkstemp()
     with open(smtp_tls_cert_fd, 'w') as f:
         f.write(f"{smtp_tls_cert}\n{smtp_tls_key}\n")
 
-tls_context = smtplib.ssl.SSLContext()
-tls_context.load_cert_chain(smtp_tls_cert_file, smtp_tls_key_file)
+tls_context = smtplib.ssl.create_default_context()
```
Comment (jkupferer) [inline notifier/operator/operator.py:66]: The change from `smtplib.ssl.SSLContext()` to `smtplib.ssl.create_default_context()` has caused some trouble. It seems that we are currently not verifying the server-side TLS cert and this switches it to check the server cert. We'll need a way to pass in a CA cert to enable this.

Comment (jkupferer) [reply notifier/operator/operator.py:66]: I successfully tested with server cert validation by adding:
```
smtp_tls_ca_cert = os.environ.get('SMTP_TLS_CA_CERT', None)
smtp_tls_ca_cert_file = os.environ.get('SMTP_TLS_CA_CERT_FILE', None)
```
and
```
tls_context = smtplib.ssl.create_default_context(
    cadata = smtp_tls_ca_cert,
    cafile = smtp_tls_ca_cert_file,
)
```

Comment (makirill) [reply notifier/operator/operator.py]: Added cert validate as described. I also updated helm chart in order to pass TLS_CA_CERT but able to test it in "dry-run" mode without real TLS CA certificate.

**Comment by jkupferer** [comment]
Merging into a branch to rework the helm chart for TLS settings

**Pull request merged**